# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/CanvasChild.cpp
# Commit: 21be69951c96
# Full Hash: 21be69951c962d3833d7117f56c2909b20b3d841
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2024-01-14 09:31:25
# Regressor Bug: 1874534
# File Overlap Count: 2
# Description:
#   Bug 1874534 - Fix DrawTargetWebgl and RemoteTextureMap context loss handling. r=aosmond
#   
#   When a context loss occurs on DrawTargetWebgl, this may result in a fallback TextureData
#   being created. Each of these are currently managed by two different RemoteTextureOwnerClients.
#   This is not really safe at all.
# ==============================================================================

diff -r 0930954b46bb -r 21be69951c96 gfx/layers/ipc/CanvasChild.cpp
--- a/gfx/layers/ipc/CanvasChild.cpp	Sat Jan 13 18:00:25 2024 +0000
+++ b/gfx/layers/ipc/CanvasChild.cpp	Sat Jan 13 18:19:24 2024 +0000
@@ -194,6 +194,7 @@
 }
 
 ipc::IPCResult CanvasChild::RecvBlockCanvas() {
+  mBlocked = true;
   if (auto* cm = gfx::CanvasManagerChild::Get()) {
     cm->BlockCanvas();
   }
@@ -492,6 +493,9 @@
 }
 
 bool CanvasChild::RequiresRefresh(int64_t aTextureId) const {
+  if (mBlocked) {
+    return true;
+  }
   auto it = mTextureInfo.find(aTextureId);
   if (it != mTextureInfo.end()) {
     return it->second.mRequiresRefresh;